lib/class-wp-rest-menu-items-controller.php

```diff
@@ -235,6 +234,56 @@ public function update_item( $request ) {
 		return rest_ensure_response( $response );
 	}
 
+	/**
+	 * Deletes a single menu item.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
```

Comment:
This delete method is basically a copy and paste of revision delete item. This method is updated to overide limitations of trash param.

Reply:
What are the limitations?

Reply:
This is a copy and paste from revisions endpoint [here](https://github.com/WordPress/WordPress/blob/master/wp-includes/rest-api/endpoints/class-wp-rest-revisions-controller.php#L443-L501). This is because both revisions and attachment do not have a trash state. Once in core, we can do something like better here.

Reply:
We should be able to use the `"rest_{$this->post_type}_trashable"` filter, no?